### DISCUSSION TITLE
Curation User Stories

### DISCUSSION THREAD

**Discussion by aj-maz** [dcgtc/dgrants#618] [Architecture & Design]
- Company "A" wants his CEO to be able to curate the grants for their own deployment of dgrants explorer ( curation list a )
- DAO "B" wants their token holders to be able to decide which grants should be visible on their dgrants explorer ( curation list b )
- DAO "C" wants that the Kleros curation system decide which grants should be visible on their dgrants explorer ( curation list c )
- Gitcoin FDD workstream wants their custom made decentralized curation system decides which grants should be visible on the main dgrants explorer ( curation list d )
- Ethereum foundation wants to run their own dgrants explorer and they want to reuse the FDD curation system (curation list d)
- Gitcoin FDD found a bug in their decentralized curation system so they want to update their system ( curation list e)

There must be a way to use all curation lists onChain to set "Grant Round Grant List" but I think it could be an off-chain tool that will filter the list and put them as a parameter in grant registery.

**Comment by phutchins** [comment]
Questions that I think we need to answer
1) Should the curated list of grants live in an on chain contract?
2) Is there ever a need to have this curated list live off chain?

It seems like a curation contract could exist that would have simple setters and getters to add and remove grant ids to and from the list that lives in that contract.

Currently the Grant Round Grant List lives in metadata in IPFS but we likely should make it such that you could use either metadata on IPFS or a contract on chain. This contract could be identical to the one described above.

**Comment by aj-maz** [comment]
I think we need to have the basic lists on-chain , and all other metadata offchain. I think the Curator contracts could be very different but we gonna need a common interface that must be implemented so our own contracts gonna interact with this interface. I suggest two solutions, both solutions have two main functions: getCuratedGrants and isCurated. 

The Curator1 interface is simpler but the Curator2 gives us the flexibility to have a multiple curation list within a single Curator contract for different categories. Also, the Curator2 has another function that will return the list of categories.

```
   interface Curator1 {
       function getCuratedGrants() external view returns (uint96[] memory);
       function isCurated(uint96) external view returns (bool);
   }
```
```
interface Curator2 {
    function getCategories() external view returns (string[] memory);
    function getCuratedGrants(string memory category )  external view returns (uint96[] memory);
    function isCurated(string memory category , uint96 id) external view returns (bool);
}
```
